lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -311,10 +330,16 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
         // if we are generating multiple parsers from the same file:
         rust!(self.out, "#[allow(dead_code)]");
         rust!(self.out,
-              "pub enum {}Symbol<{}> {{",
+              "pub enum {}Symbol<{}>",
```

Comment:
I wonder... so currently we only include those lifetime/type-parameters that we *need* -- but maybe it'd be better, or at least easier, to include *all* of them, and just throw in a phantom data. Then we wouldn't have to scrape around so much to find out which ones are actually used, right?

Reply:
Would all these symbols be in a a new, unused variant? That could have the unfortunate side effect of making the discriminate larger, right?

I guess that seems like a fine trade off if the finding of lifetime/type parameters is a code/maintenance burden and has a history of bugs. If not, then it doesn't seem worth it. What do you think?

Reply:
Yes, it'd make the discriminant larger. That seems like a non-issue to me. But let's just merge this PR for now, I think, and we can revisit later.